components/focus-trap/focus-trap-mixin.js

```diff
@@ -0,0 +1,99 @@
+import { getNextFocusable, getPreviousFocusable } from '../../helpers/focus.js';
+import { html } from 'lit-element/lit-element.js';
+import { isComposedAncestor } from '../../helpers/dom.js';
+
+export const FocusTrapMixin = superclass => class extends superclass {
+
+	static get properties() {
+		return {
+			trap: { type: Boolean, reflect: true }
+		};
+	}
+
+	constructor() {
+		super();
+		this.trap = false;
+		this._handleFocusTrapBodyFocus = this._handleFocusTrapBodyFocus.bind(this);
+	}
+
+	async attributeChangedCallback(name, oldval, newval) {
+		super.attributeChangedCallback(name, oldval, newval);
+		if (name !== 'trap') return;
+		if (this.trap) {
+			document.body.addEventListener('focus', this._handleFocusTrapBodyFocus, true);
+		} else {
+			document.body.removeEventListener('focus', this._handleFocusTrapBodyFocus, true);
+		}
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		if (this.trap) {
+			document.body.removeEventListener('focus', this._handleFocusTrapBodyFocus, true);
+		}
+	}
+
+	_focusTrapFocusFirst() {
+		if (this._focusFirst) {
+			this._focusFirst();
+			return;
+		}
+		const focusable = getNextFocusable(this.shadowRoot.querySelector('.d2l-focus-trap-start'));
+		if (focusable) focusable.focus();
+	}
+
+	_getFocusTrapContainer() {
+		return this.shadowRoot.querySelector('.d2l-focus-trap-start').parentNode;
+	}
+
+	_handleFocusTrapBodyFocus(e) {
+		const trapContainer = this._getFocusTrapContainer();
```

Comment:
Perhaps I would check `this.trap` here as a precaution... even though this handler should not be wired up while not trapping.